### PR TITLE
Token management rebase fix

### PR DESCRIPTION
**Pull request by maranai**

### CODE REVIEW

File: impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultPasswordGrantRequestBuilder.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    @Override
+    public PasswordGrantRequestBuilder setLogin(String login) {
+        Assert.hasText(login, "username cannot be null or empty.");
+        this.login = login;
+        return this;
+    }
+
+    @Override
+    public PasswordGrantRequestBuilder setPassword(String password) {
+        Assert.notNull(password, "password cannot be null or empty.");
```
Comment (dogeared) [inline impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultPasswordGrantRequestBuilder.java]: Recreating comment from @mrioan in prior PR: 

Not sure if the builder should restrict passwords not to be null... Similar classes (like DefaultUsernamePasswordRequestBuilder) do allow passwords to be null

Comment (dogeared) [inline impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultPasswordGrantRequestBuilder.java]: Recreating response from @maranai in prior PR:

@mrioan if a password is not provided the token exchange request sent to AM will return an exception ("password parameter cannot be null, empty, or blank."). Not sure what our preference for AM exceptions is, do we alert users when required data is missing or just let the request flow continue?

File: impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultPasswordGrantRequestBuilder.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    public PasswordGrantRequestBuilder setAccountStore(AccountStore accountStore) {
+        Assert.notNull(accountStore, "accountStore cannot be null or empty.");
+        this.accountStore = accountStore;
+        return this;
+    }
+
+    @Override
+    public PasswordGrantRequest build() {
+        Assert.state(this.login != null, "login has not been set. It is a required attribute.");
+        Assert.state(this.password != null, "password has not been set. It is a required attribute.");
```
Comment (dogeared) [inline impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultPasswordGrantRequestBuilder.java]: Recreating comment from @mrioan in prior PR:

Idem comment.

**Comment by dogeared** [comment]
@maranai @mrioan - There was [another PR](https://github.com/stormpath/stormpath-sdk-java/pull/291) that got corrupted during a rebase/merge attempt.

This is the cleaned up PR of the same code changes that @maranai did for this feature.

I recreated the only outstanding comments from @mrioan on the original. Every other comment was already addressed.

File: api/src/main/java/com/stormpath/sdk/oauth/Authenticators.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.stormpath.sdk.oauth;
+
+import com.stormpath.sdk.lang.Classes;
+
+/**
+ * Static utility/helper methods serving {@link com.stormpath.sdk.oauth.Oauth2AuthenticatorFactory Oauth2AuthenticatorFactory}s. For example, to
```
Comment (mrioan) [inline api/src/main/java/com/stormpath/sdk/oauth/Authenticators.java]: Static utility/helper class serving

File: api/src/main/java/com/stormpath/sdk/oauth/JwtAuthenticator.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+package com.stormpath.sdk.oauth;
+
+/**
+ * This class is used to authenticate a Json Web Token against Stormpath. For example:
+ * <pre>
+ * Application app = obtainApplication();
+ * JwtAuthenticationRequest authRequest = Oauth2Requests.JWT_AUTHENTICATION_REQUEST
+ *      .builder()
+ *      .setJwt(jwt)
+ *      .build()
```
Comment (mrioan) [inline api/src/main/java/com/stormpath/sdk/oauth/JwtAuthenticator.java]: missing semicolon: `.build();`

File: api/src/main/java/com/stormpath/sdk/oauth/Oauth2Requests.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+*/
+package com.stormpath.sdk.oauth;
+
+import com.stormpath.sdk.lang.Classes;
+
+/**
+ * Static utility/helper methods serving Grant Authentication Request Builders. For example, to
+ * construct a builder to create a {@link PasswordGrantRequest}:
+ * <pre>
+ * PasswordGrantRequest request = Authenticators.PASSWORD_GRANT_REQUEST.builder()
```
Comment (mrioan) [inline api/src/main/java/com/stormpath/sdk/oauth/Oauth2Requests.java]: `Authenticators.PASSWORD_GRANT_REQUEST.builder()` should be `Oauth2Requests.PASSWORD_GRANT_REQUEST.builder()`

File: api/src/main/java/com/stormpath/sdk/oauth/Oauth2Requests.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+ *      .setLogin(username)
+ *      .setPassword(password)
+ *      .build();
+ * </pre>
+ * or, to construct a builder for a Refresh Grant Authentication request:
+ * <pre>
+ * RefreshGrantRequest request = Authenticators.REFRESH_GRANT_REQUEST.builder()
```
Comment (mrioan) [inline api/src/main/java/com/stormpath/sdk/oauth/Oauth2Requests.java]: `Authenticators.REFRESH_GRANT_REQUEST.builder()` should be `Oauth2Requests.REFRESH_GRANT_REQUEST.builder()`

File: api/src/main/java/com/stormpath/sdk/oauth/PasswordGrantAuthenticator.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+* limitations under the License.
+*/
+package com.stormpath.sdk.oauth;
+
+/**
+ * Interface denoting a Password Grant-specific {@link Oauth2Authenticator}. It is used to authenticate an account and
+ * exchange its credentials for a valid OAuth 2.0 token. For example:
+ * <pre>
+ * Application app = obtainApplication();
+ * PasswordGrantRequest request = <b>Authenticators.PASSWORD_GRANT_REQUEST.builder()</b>
```
Comment (mrioan) [inline api/src/main/java/com/stormpath/sdk/oauth/PasswordGrantAuthenticator.java]: `Authenticators.PASSWORD_GRANT_REQUEST.builder()` must be `Oauth2Requests.PASSWORD_GRANT_REQUEST.builder()`

File: api/src/main/java/com/stormpath/sdk/oauth/RefreshGrantAuthenticator.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package com.stormpath.sdk.oauth;
+
+/**
+ * This class is used to refresh an OAuth 2.0 token created in Stormpath. For example:
+ * <pre>
+ * Application app = obtainApplication();
+ * RefreshGrantRequest request = <b>Authenticators.REFRESH_GRANT_REQUEST.builder()</b>
```
Comment (mrioan) [inline api/src/main/java/com/stormpath/sdk/oauth/RefreshGrantAuthenticator.java]: `Authenticators.REFRESH_GRANT_REQUEST.builder()` should be `Oauth2Requests.REFRESH_GRANT_REQUEST.builder()`

**Comment by mrioan** [comment]
Back to you @maranai . If you agree with all the changes, you can consider this PR approved after the changes have been applied. Thanks!

**Pull request merged**